sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/owaspzapwrapper/config/ApiDefinitionFileProviderTest.java

```diff
@@ -0,0 +1,134 @@
+    }
+
+    @Test
+    void missing_filesystem_part_results_in_must_exit_runtime_exception() {
+        /* prepare */
+        String sechubScanConfigJSON = "{\"apiVersion\":\"1.0\",\"data\":{\"sources\":[{\"name\":\"open-api-file-reference\"}]},"
+                + "\"webScan\":{\"uri\":\"https://localhost:8443\",\"api\":{\"type\":\"openApi\",\"use\":[\"open-api-file-reference\"]}}}";
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute + test */
+        MustExitRuntimeException exception = assertThrows(MustExitRuntimeException.class,
+                () -> providerToTest.fetchApiDefinitionFile("/example/path/to/extracted/sources", sechubScanConfiguration));
+
+        assertEquals("Sources filesystem part must be set at this stage.", exception.getMessage());
+        assertEquals(MustExitCode.SECHUB_CONFIGURATION_INVALID, exception.getExitCode());
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("filesystemPartSizeTestNamedArguments")
+    void filesystem_part_with_size_other_than_exactly_one_results_in_must_exit_runtime_exception(String sechubScanConfigJSON) {
+        /* prepare */
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute + test */
+        MustExitRuntimeException exception = assertThrows(MustExitRuntimeException.class,
+                () -> providerToTest.fetchApiDefinitionFile("/example/path/to/extracted/sources", sechubScanConfiguration));
+
+        assertEquals("Sources filesystem files part must contain exactly 1 entry.", exception.getMessage());
+        assertEquals(MustExitCode.SECHUB_CONFIGURATION_INVALID, exception.getExitCode());
+    }
+
+    @Test
+    void provider_returns_correct_path_for_valid_sechub_scan_config_with_openapi_definition_file() {
+        /* prepare */
+        String sechubScanConfigJSON = "{\"apiVersion\":\"1.0\",\"data\":{\"sources\":[{\"name\":\"open-api-file-reference\",\"fileSystem\":{\"files\":[\"openapi3.json\"]}}]},"
+                + "\"webScan\":{\"uri\":\"https://localhost:8443\",\"api\":{\"type\":\"openApi\",\"use\":[\"open-api-file-reference\"]}}}";
+        SecHubScanConfiguration sechubScanConfiguration = SecHubScanConfiguration.createFromJSON(sechubScanConfigJSON);
+
+        /* execute */
+        Path path = providerToTest.fetchApiDefinitionFile("test/path", sechubScanConfiguration);
+
+        /* test */
+        assertEquals(path.toString(), "test/path/openapi3.json");
+    }
+
+    static Stream<Arguments> sourcesPartSizeTestNamedArguments() {
+        /* @formatter:off */
+        return Stream.of(
+        		Arguments.of(Named.of("Sources part empty", "{\"apiVersion\":\"1.0\",\"data\":{\"sources\":[]},\"webScan\":{\"uri\":\"https://localhost:8443\",\"api\":{\"type\":\"openApi\",\"use\":[\"open-api-file-reference\"]}}}")),
```

Comment:
This is hard to read. How about writing it like this:

~~~
Arguments.of(
  Named.of( "Sources part empty",
            "{\"apiVersion\":\"1.0\",\"data\":{\"sources\":[]},\"webScan\":{\"uri\":\"https://localhost:8443\",\"api\":{\"type\":\"openApi\",\"use\":[\"open-api-file-reference\"]}}}"
  )
),
~~~

Or something similar.